### PR TITLE
Fix empty HID list introduced by #1408, uniformise select behavior

### PR DESCRIPTION
**Pull request by p3r7**
my bad

### CODE REVIEW

**Comment by p3r7** [comment]
arf, and the hid select is kinda borked when there are many items.

dunno if it's a regression or not, gonna clean up the entry selection code.

File: lua/core/menu/devices.lua [hunk 1]
```diff
@@ -130,25 +133,41 @@ m.enc = function(n,delta)
   end
 end
 
-local function redraw_keyboard_layout_select()
-  local len = tabutil.count(m.options["keyboard layout"])
+local function redraw_select(do_show_all_if_fit, do_uppercase)
```
Comment (p3r7) [inline lua/core/menu/devices.lua:136]: this fn basically implements your comment https://github.com/monome/norns/pull/1408#issuecomment-909635580

File: lua/core/menu/devices.lua [hunk 2]
```diff
@@ -130,25 +133,41 @@ m.enc = function(n,delta)
...
+
   screen.update()
 end
 
 m.redraw = function()
-
-  if (m.section == "keyboard layout" and m.mode == "select") then
-    redraw_keyboard_layout_select()
+  if m.mode == "select" then
+    local do_show_all_if_fit = false
```
Comment (p3r7) [inline lua/core/menu/devices.lua:168]: i disabled the "show all if less than 6 items" (https://github.com/monome/norns/pull/1408#issuecomment-909635580) as the absence of top padding forces a cognitive shift.

(in my brain i associated the top padding w/ "oh, i'm in the select screen" and i'm kinda lost w/ its absence)

you can play w/ the value to see for yourself.

**Review by tehn — APPROVED** [review]
thanks for the attention to detail

**Pull request merged**